### DISCUSSION TITLE
Feature Request - Password/link read access to private repositories

### DISCUSSION THREAD

**Discussion by jbellUIC** [community/community#32813] [Code Security]
I would like to request a feature by which access can be given to a private repository via a link or password, so that anyone with the link or password could read the repository, but it would still remain private otherwise.

My students create repositories for their projects, which I would like to keep private so that other students can't see or copy their work.  On the other hand, the students would like to give prospective employers access to their repositories as examples of their work.  I would like for the students to put on their resume a link or password that allows prospective employers read access to the repositories while keeping them private for other students.

**Comment by itcarroll** [comment]
See also [44506](https://github.com/orgs/community/discussions/44506).

**Comment by Truclam104** [comment]
> I would like to request a feature by which access can be given to a private repository via a link or password, so that anyone with the link or password could read the repository, but it would still remain private otherwise.
> 
> My students create repositories for their projects, which I would like to keep private so that other students can't see or copy their work.  On the other hand, the students would like to give prospective employers access to their repositories as examples of their work.  I would like for the students to put on their resume a link or password that allows prospective employers read access to the repositories while keeping them private for other students.

https://www.facebook.com/profile.php?id=100004393635904&mibextid=ZbWKwL

**Comment by Chris-C3A** [comment]
This [https://gitshare.me](https://gitshare.me) solves exactly this problem!
